torch/_inductor/triton_ops/autotune.py

```diff
@@ -80,11 +79,13 @@ def _precompile_config(self, cfg: Config, warm_cache_only_with_cc: int):
             return
 
         torch.cuda.set_device(torch.cuda.current_device())
-
-        binary = triton.compile(
-            self.fn,
-            **compile_meta,
-        )
+        with torch.cuda.device(compile_meta["device"]):
+            # need to initialize context
+            torch.cuda.synchronize(torch.cuda.current_device())
```

Comment:
why is this needed?

Reply:
Otherwise `triton.compile` errors out with "context not initialized", which should properly be fixed on triton side, but this will do for now.

Reply:
@wconstab @ngimel Are we referring to the cuda context here or some other context? If so, I'm assuming it should be possible to first check if the context is initialized and only do this if the context is not initialized? See https://github.com/pytorch/pytorch/issues/109074 for context